From df0fd823551a8daea918118f26dac217f1f2d4da Mon Sep 17 00:00:00 2001
From: Vignesh Goutham Ganesh <vgg@amazon.com>
Date: Tue, 11 Jan 2022 21:05:13 -0800
Subject: [PATCH 01/10] OVA improvements

- Create /etc/pki/tls/certs dir as part of image-builds
- Tweak Product info in OVF

Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 images/capi/ansible/roles/sysprep/tasks/main.yml |  9 +++++++++
 images/capi/hack/ovf_template.xml                | 10 ++--------
 2 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/images/capi/ansible/roles/sysprep/tasks/main.yml b/images/capi/ansible/roles/sysprep/tasks/main.yml
index a9fa954d5..a526528ea 100644
--- a/images/capi/ansible/roles/sysprep/tasks/main.yml
+++ b/images/capi/ansible/roles/sysprep/tasks/main.yml
@@ -27,6 +27,15 @@
 - import_tasks: photon.yml
   when: ansible_os_family == "VMware Photon OS"
 
+- name: Get stats of the /etc/pki folder
+  ansible.builtin.stat:
+    path: /etc/pki
+  register: pki
+
+- name: Ensures /etc/pki/tls/certs dir exists
+  file: path=/etc/pki/tls/certs state=directory
+  when: pki.stat.isdir is defined and pki.stat.isdir
+
 - name: Remove containerd http proxy conf file if needed
   file:
     path: /etc/systemd/system/containerd.service.d/http-proxy.conf
diff --git a/images/capi/hack/ovf_template.xml b/images/capi/hack/ovf_template.xml
index 316427ec3..ca23db5f9 100644
--- a/images/capi/hack/ovf_template.xml
+++ b/images/capi/hack/ovf_template.xml
@@ -147,19 +147,13 @@
     <vmw:BootOrderSection vmw:instanceId="8" vmw:type="disk">
       <Info>Virtual hardware device boot order</Info>
     </vmw:BootOrderSection>
-    <EulaSection>
-      <Info>An end-user license agreement</Info>
-      <License>
-${EULA}
-      </License>
-    </EulaSection>
     <ProductSection>
       <Info>Information about the installed software</Info>
       <Product>${PRODUCT}</Product>
-      <Vendor>VMware Inc.</Vendor>
+      <Vendor>AWS</Vendor>
       <Version>${TYPED_VERSION}</Version>
       <FullVersion>${TYPED_VERSION}</FullVersion>
-      <VendorUrl>https://vmware.com</VendorUrl>
+      <VendorUrl>https://aws.amazon.com/</VendorUrl>
       <Category>Cluster API Provider (CAPI)</Category>
       <Property ovf:userConfigurable="false" ovf:value="${BUILD_TIMESTAMP}" ovf:type="string" ovf:key="BUILD_TIMESTAMP"/>
       <Property ovf:userConfigurable="false" ovf:value="${BUILD_DATE}" ovf:type="string" ovf:key="BUILD_DATE"/>
-- 
2.39.3 (Apple Git-146)

